### PR TITLE
Updating OB OpenAPI Taxonomy Checker, Fixing Errors/Warnings It Raises About Taxonomy

### LINKED ISSUE #62: Add "Diamond / Diagonal" enum to moduleOrientationItemType
Example: https://www.sunstyle.com/

### PR DESCRIPTION
**Pull request by reepoi**
Closes #62, #53, #33, #26, #12

This updates the IBM OpenAPI Validator version from 0.52.1 to 0.76.2, adds new validation rules, and fixes most of the errors/warnings raised by the taxonomy checker.

This PR includes the changes in #17, so #17 should be merged before this PR.

To update the version of IBM OpenAPI Validator, these changes were made:
- workflows/schema-validator.yaml now installs IBM's OpenAPI Ruleset package (@ibm-cloud/openapi-ruleset) so that the OB Taxonomy checker can extend and customize it using the config file .spectral.yaml file.
- In workflows/schema-validator.yaml, the version of the IBM OpenAPI Validator is pinned to 0.76.2, and @ibm-cloud/openapi-ruleset is pinned to 0.25.2.
- All rules in .validaterc were deprecated, so .validaterc was deleted. The rules it provided are now included in @ibm-cloud/openapi-ruleset.
- The IBM validator was modified so that its rules related to schema definitions would check the schema definitions in the components section of the OB OpenAPI taxonomy JSON. This modification was done by wrapping the IBM validator with a new custom validator. See ibm-oas-wrapper-check-components.js for details. The config file .spectral.yaml now extends from this new validator.
- The IBM validator rules enforcing case conventions are turned off (property-case-convention, and enum-case-convention).
- The rules major-version-in-path, openapi-tags, oas3-api-servers, and oas3-unused-component emit warnings by default in the IBM validator configuration. Listing them in .spectral.yaml is repetitive, so these lines were removed.

The new validation rules are:
- Every taxonomy element has a valid item type
- Every taxonomy element has a valid item type group or no item type group
- Every taxonomy element has a valid sample value or no sample value (x-ob-sample-value is an empty object)
- The OpenAPI type of taxonomy elements matches the type suggested by their OB item type
- Enums/Units listed in an item type group exist in the definition of the item type

To fix the errors/warnings raised by the taxonomy checker, these changes were made to the taxonomy:
- Item type prefixes are removed
- the casing of item type names and the names of their enums match the OB case convention
- new substrings and words were added to the spellchecker dictionary OB-words

### CODE REVIEW

**Comment by cwhanse** [comment]
Now I recall why we may have had the acronyms and abbreviations separate: we were going to spell out what each meant. Is it possible to do that in the .dic file?  Maybe separate the acronym from its definition by a specific character - it looks like "/" is [reserved ](https://github.com/cfinke/Typo.js/blob/9955cf1ece1897d7e585eb03ff7cc64eb3935a41/typo/typo.js#L446)for that purpose.

**Comment by reepoi** [comment]
I can't find a way to add comments to words in a .dic file, but there are some workarounds. If we add two "/" after a word, e.g. "ACH//Automated Clearing House", Typo.js will skip trying to create any spelling rules for variants. We can also store the words and their meanings in a different file format like JSON, and then have the validator dynamically create the .dic file before running the spellchecker.
I will make a list of the meanings of each word in the .dic file, too.

File: .github/configs/schema-validator/.spectral.yaml [hunk 1]
```diff
@@ -1,11 +1,89 @@
...
+    message: 'Item type group does not exist: {{error}}'
+    given: $.components.schemas[*].allOf[1][x-ob-item-type-group]
+    severity: error
+    then:
+      function: validateXOBFields
+      functionOptions:
+        objectPath: x-ob-item-type-groups
 
+  item-type-group-valid-type:
+    message: 'Item type group type does not exist: {{error}}'
```
Comment (cwhanse) [inline .github/configs/schema-validator/.spectral.yaml:37]:
```suggestion
    message: 'Item type group references an item type that does not exist: {{error}}'
```

File: .github/configs/schema-validator/.spectral.yaml [hunk 2]
```diff
@@ -1,11 +1,89 @@
...
+    given: $[x-ob-item-type-groups][*].type
+    severity: error
+    then:
+      function: refPathExists
+  
+  item-type-group-valid-entries:
+    description: The enums or units in the item type group exist in the item type definition.
+    message: '{{error}}'
```
Comment (cwhanse) [inline .github/configs/schema-validator/.spectral.yaml:45]: Should there be specific text in the message?

File: .github/configs/schema-validator/.spectral.yaml [hunk 3]
```diff
@@ -1,11 +1,89 @@
...
+    then:
+      function: validateItemTypeGroupEntries
+  
+  element-valid-sample-value:
+    description: >-
+      Checks the following items about sample value:
+      - It is an object. The object may be empty (see requireAtLeastOneField).
+      - Its fields are only primitives.
+      - Value has the correct OpenAPI type
+      - If the schema definition OB item type defines enums or units, and it has an item type group:
```
Comment (cwhanse) [inline .github/configs/schema-validator/.spectral.yaml:57]: Is the check for valid enum or unit skipped if an item type group is not present?

File: .github/configs/schema-validator/.spectral.yaml [hunk 4]
```diff
@@ -1,11 +1,89 @@
...
+        # Require sample value be a nonempty object
+        requireAtLeastOneField: false
+        enumItemTypeIgnoreList:
+          - UUIDItemType # Value is checked against the regex of UUID
+
+  element-valid-openapi-type-and-item-type:
+    description: >-
+      Check that the element's OpenAPI type and OB item type are compatible.
+      OpenAPI 'integer' and 'number' types are only compatible with OB unit item types.
+      OpenAPI 'boolean' and 'string' types are only compatible with OB enum item types.
```
Comment (cwhanse) [inline .github/configs/schema-validator/.spectral.yaml:78]: All OB enums are strings, but I don't think OpenAPI boolean are strings: they are literals true and false.

**Review by cwhanse — APPROVED** [review]

**Review by jrippingale — APPROVED** [review]
Wow - fabulous upgrade!  Thank you!

File: .github/configs/schema-validator/.spectral.yaml (hunk 1 above)
Comment (reepoi) [reply .github/configs/schema-validator/.spectral.yaml]: Ok, I made this change.

File: .github/configs/schema-validator/.spectral.yaml (hunk 3 above)
Comment (reepoi) [reply .github/configs/schema-validator/.spectral.yaml]: The enum or unit sample values are always checked against the item type. If an item type group is present, the enum or unit sample values must also be in the item type group. I updated the validation rule description to describe that.

**Pull request merged**